File: backend/scripts/replace_getCurrentUserId.py
"""
全コントローラーの getCurrentUserId() を SecurityUtils.getCurrentUserId() に置換するスクリプト。

処理内容:
1. TODO コメント + getCurrentUserId() メソッド定義を削除
2. getCurrentUserId() の呼び出しを SecurityUtils.getCurrentUserId() に置換
3. import com.mannschaft.app.common.SecurityUtils を追加
"""

import re
import glob
import os

SRC_ROOT = os.path.join(os.path.dirname(__file__), "..", "src", "main", "java")
IMPORT_LINE = "import com.mannschaft.app.common.SecurityUtils;\n"

# getCurrentUserId メソッド定義のパターン（TODO コメント + メソッド本体）
METHOD_PATTERN = re.compile(
    r'\n'
    r'(    // TODO:.*(?:SecurityContextHolder|SecurityContext|認証情報|セキュリティコンテキスト|JWT).*\n)?'
    r'    private Long getCurrentUserId\(\) \{\n'
    r'        return 1L;\n'
    r'    \}\n',
    re.MULTILINE
)

# SecurityUtils.java 自身は除外
EXCLUDE_FILES = {"SecurityUtils.java"}

def process_file(filepath):
    with open(filepath, "r", encoding="utf-8") as f:
        content = f.read()

    if "getCurrentUserId" not in content:
        return False

    original = content

    # 1. メソッド定義を削除
    content = METHOD_PATTERN.sub("\n", content)

    # 2. getCurrentUserId() 呼び出しを置換
    content = content.replace("getCurrentUserId()", "SecurityUtils.getCurrentUserId()")

    # 3. import 追加（まだない場合）
    if "import com.mannschaft.app.common.SecurityUtils;" not in content:
        # 最後の import 行の後に追加
        last_import_idx = content.rfind("\nimport ")
        if last_import_idx != -1:
            # 行末を見つける
            line_end = content.index("\n", last_import_idx + 1)
            content = content[:line_end + 1] + IMPORT_LINE + content[line_end + 1:]

    if content != original:
        with open(filepath, "w", encoding="utf-8") as f:
            f.write(content)
        return True
    return False

def main():
    pattern = os.path.join(SRC_ROOT, "com", "mannschaft", "app", "**", "*.java")
    files = glob.glob(pattern, recursive=True)

    # worktree 内のファイルは除外
    files = [f for f in files if ".claude" not in f and os.path.basename(f) not in EXCLUDE_FILES]

    modified_count = 0
    modified_files = []

    for filepath in sorted(files):
        if process_file(filepath):
            modified_count += 1
            modified_files.append(filepath)

    print(f"Modified {modified_count} files:")
    for f in modified_files:
        # 相対パスで表示
        rel = os.path.relpath(f, os.path.join(os.path.dirname(__file__), ".."))
        print(f"  {rel}")

if __name__ == "__main__":
    main()

File: backend/scripts/replace_getCurrentUserId.mjs
/**
 * 全コントローラーの getCurrentUserId() を SecurityUtils.getCurrentUserId() に置換するスクリプト。
 *
 * 処理内容:
 * 1. TODO コメント + getCurrentUserId() メソッド定義を削除
 * 2. getCurrentUserId() の呼び出しを SecurityUtils.getCurrentUserId() に置換
 * 3. import com.mannschaft.app.common.SecurityUtils を追加
 */

import { readFileSync, writeFileSync } from 'fs';
import { glob } from 'fs/promises';
import { resolve, relative, basename } from 'path';

const SRC_ROOT = resolve(import.meta.dirname, '..', 'src', 'main', 'java');
const IMPORT_LINE = 'import com.mannschaft.app.common.SecurityUtils;';
const EXCLUDE = new Set(['SecurityUtils.java']);

// getCurrentUserId メソッド定義パターン（TODOコメント + 本体）
const METHOD_RE = /\n(    \/\/ TODO:.*(?:SecurityContextHolder|SecurityContext|認証情報|セキュリティコンテキスト|JWT|JwtAuthentication).*\n)?    private Long getCurrentUserId\(\) \{\n        return 1L;\n    \}\n/g;

let modifiedCount = 0;
const modifiedFiles = [];

const pattern = resolve(SRC_ROOT, 'com/mannschaft/app/**/*.java');

for await (const filepath of glob(pattern)) {
  if (filepath.includes('.claude') || EXCLUDE.has(basename(filepath))) continue;

  let content = readFileSync(filepath, 'utf-8');
  if (!content.includes('getCurrentUserId')) continue;

  const original = content;

  // 1. メソッド定義を削除
  content = content.replace(METHOD_RE, '\n');

  // 2. 呼び出しを置換
  content = content.replaceAll('getCurrentUserId()', 'SecurityUtils.getCurrentUserId()');

  // 3. import 追加
  if (!content.includes(IMPORT_LINE)) {
    const lastImportIdx = content.lastIndexOf('\nimport ');
    if (lastImportIdx !== -1) {
      const lineEnd = content.indexOf('\n', lastImportIdx + 1);
      content = content.slice(0, lineEnd + 1) + IMPORT_LINE + '\n' + content.slice(lineEnd + 1);
    }
  }

  if (content !== original) {
    writeFileSync(filepath, content, 'utf-8');
    modifiedCount++;
    modifiedFiles.push(relative(resolve(import.meta.dirname, '..'), filepath));
  }
}

console.log(`Modified ${modifiedCount} files:`);
modifiedFiles.forEach(f => console.log(`  ${f}`));
